src/ReadLine/ReadLine.cs

```diff
@@ -31,7 +43,14 @@ public static string Read(string prompt = "")
 
             Console.WriteLine();
             string text = _keyHandler.Text;
-            _history.Add(text);
+            if (String.IsNullOrWhiteSpace(text) && !String.IsNullOrWhiteSpace(defaultInput))
+            {
+                text = defaultInput;
+            }
+            else
+            {
+                _history.Add(text);
+            }
```

Comment:
As it is written "text" would be an empty string, I felt like adding the default to the history didn't make sense, and adding an empty value also would not make sense, and that is why I omit adding anything to history if the default value is used.  If we were going to go with 1 of the 2 options, adding the default value would be the better choice, but then hitting (up) would have something that you never typed in.